dotCMS/src/main/java/com/dotcms/analytics/track/AnalyticsTrackWebInterceptor.java

```diff
@@ -0,0 +1,105 @@
+
+/**
+ * Web Interceptor to track analytics
+ * @author jsanca
+ */
+public class AnalyticsTrackWebInterceptor  implements WebInterceptor {
+
+    private final static Map<String, RequestMatcher> requestMatchersMap = new ConcurrentHashMap<>();
+
+    private final EventLogSubmitter submitter;
+
+    /// private static final String[] DEFAULT_BLACKLISTED_PROPS = new String[]{"^/api/*"};
+    private static final String[] DEFAULT_BLACKLISTED_PROPS = new String[]{StringPool.BLANK};
+    private final WhiteBlackList whiteBlackList = new WhiteBlackList.Builder()
+            .addWhitePatterns(Config.getStringArrayProperty("ANALYTICS_WHITELISTED_KEYS",
+                    new String[]{StringPool.BLANK})) // allows everything
+            .addBlackPatterns(CollectionsUtils.concat(Config.getStringArrayProperty(  // except this
+                    "ANALYTICS_BLACKLISTED_KEYS", new String[]{}), DEFAULT_BLACKLISTED_PROPS)).build();
+
+    public AnalyticsTrackWebInterceptor() {
+
+        submitter = new EventLogSubmitter();
+        addRequestMatcher(
+                new PagesAndUrlMapsRequestMatcher(),
+                new FilesRequestMatcher(),
+                new RulesRedirectsRequestMatcher(),
+                new VanitiesRequestMatcher());
+    }
+
+    /**
+     * Add a request matchers
+     * @param requestMatchers
+     */
+    public static void addRequestMatcher(final RequestMatcher... requestMatchers) {
+        for (final RequestMatcher matcher : requestMatchers) {
+            requestMatchersMap.put(matcher.getId(), matcher);
+        }
+    }
+
+    /**
+     * Remove a request matcher by id
+     * @param requestMatcherId
+     */
+    public static void removeRequestMatcher(final String requestMatcherId) {
+
+        requestMatchersMap.remove(requestMatcherId);
+    }
+
+    @Override
+    public Result intercept(final HttpServletRequest request, final HttpServletResponse response) throws IOException {
+
+        if (whiteBlackList.isAllowed(request.getRequestURI())) {
+            final Optional<RequestMatcher> matcherOpt = this.anyMatcher(request, response, RequestMatcher::runBeforeRequest);
+            if (matcherOpt.isPresent()) {
+
+                Logger.debug(this, () -> "intercept, Matched: " + matcherOpt.get().getId() + " request: " + request.getRequestURI());
+                //fireNextStep(request, response);
+            }
+        }
+
+        return Result.NEXT;
+    }
+
+    @Override
+    public boolean afterIntercept(final HttpServletRequest request, final HttpServletResponse response) {
+
+        if (whiteBlackList.isAllowed(request.getRequestURI())) {
+            final Optional<RequestMatcher> matcherOpt = this.anyMatcher(request, response, RequestMatcher::runAfterRequest);
+            if (matcherOpt.isPresent()) {
+
+                Logger.debug(this, () -> "afterIntercept, Matched: " + matcherOpt.get().getId() + " request: " + request.getRequestURI());
+                //fireNextStep(request, response);
```

Comment:
are you going method to be executed for each RequestMatcher after and before intercept?
or this next steps is trigger the Jitsu Event?

Reply:
the request matcher has a method that tells to the Interceptor if should run before or after or both